src/main/java/org/sunbird/hubservices/daoimpl/GraphDao.java

```diff
@@ -61,72 +59,85 @@ public Boolean upsertNode(Node node) throws Exception {
-                if (session != null && transaction != null) {
-                    transaction.close();
-                    session.close();
-                }
-            } catch (Exception e) {
-                logger.error("Session Closed : ", e);
-                return Boolean.FALSE;
-            }
         }
-
         return Boolean.TRUE;
     }
 
     @Override
     public Boolean upsertRelation(Node nodeFrom, Node nodeTo, Map<String, String> relationProperties) throws Exception {
-        try (Session session = neo4jDriver.session()) {
-            try (Transaction transaction = session.beginTransaction()) {
-                Map<String, Object> parameters = new HashMap<>();
-                parameters.put("fromUUID", nodeFrom.getId());
-                parameters.put("toUUID", nodeTo.getId());
-                parameters.put(Constants.Graph.PROPS.getValue(), relationProperties);
+        boolean isUpserted = Boolean.FALSE;
+        try (Session session = neo4jDriver.session(); Transaction transaction = session.beginTransaction()) {
+            Map<String, Object> parameters = new HashMap<>();
+            parameters.put("fromUUID", nodeFrom.getId());
+            parameters.put("toUUID", nodeTo.getId());
+            parameters.put(Constants.Graph.PROPS.getValue(), relationProperties);
 
-                String queryNodeExistWithReverseEdge = "MATCH (n:" + label + ")<-[r:connect]-(n1:" +
-                        label + ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "RETURN n,n1";
-
-                Statement statement = new Statement(queryNodeExistWithReverseEdge, parameters);
-                StatementResult result = transaction.run(statement);
-                int recordSize = result.list().size();
-                result.consume();
-                if (recordSize != 0) {
-                    throw new GraphException(ErrorCode.NODES_WITH_REVERSE_EDGE_EXISTS.name(),
-                            "Nodes exists with reverse edge, new relation cannot be created!");
-                }
+            String queryNodeExistWithReverseEdge = "MATCH (n:" + label + ")<-[r:connect]-(n1:" +
+                    label + ") WHERE n.id = $fromUUID AND n1.id = $toUUID " + "RETURN n,n1";
 
+            Statement statement = new Statement(queryNodeExistWithReverseEdge, parameters);
+            StatementResult result = transaction.run(statement);
+            int recordSize = result.list().size();
+            result.consume();
+            if (recordSize != 0) {
+                if (logger.isDebugEnabled())
+                    logger.debug(nodeFrom.getId(), nodeTo.getId());
+                updateRelationshipBetweenTwoNodes(nodeFrom, nodeTo, statement, result, transaction, recordSize, relationProperties);
+                transaction.commitAsync().toCompletableFuture().get();
+                logger.info("user relation with toUUID {} and fromUUID {} updated successfully ", nodeTo.getId(),
+                        nodeFrom.getId());
```

Comment:
There is already an Info log printed inside both create and update relation... this is not required... if you need more info, use the Debug statement instead of info logs
when logging error make sure that it gives details of from and to UUIDs